plugins/catalog-backend/src/service/CatalogBuilder.ts

```diff
@@ -577,4 +579,80 @@ export class CatalogBuilder {
       );
     }
   }
+
+  // TODO(freben): This can be removed no sooner than June 2022, after adopters have had some time to adapt to the new package structure
+  private checkMissingExternalProcessors(processors: CatalogProcessor[]) {
+    const locationTypes = new Set(
+      this.env.config
+        .getOptionalConfigArray('catalog.locations')
+        ?.map(l => l.getString('type')) ?? [],
+    );
+    const processorNames = new Set(processors.map(p => p.getProcessorName()));
+
+    function check(
+      locationType: string,
+      processorName: string,
+      installationUrl: string,
+    ) {
+      if (
+        locationTypes.has(locationType) &&
+        !processorNames.has(processorName)
+      ) {
+        throw new Error(
+          `Your config contains a "catalog.locations" entry of type ${locationType}, \
+           but does not have the corresponding catalog processor ${processorName} installed. \
+           This processor used to be built into the catalog itself, but is now moved to an \
+           external module that has to be installed manually. Please follow the installation \
+           instructions at ${installationUrl} if you are using this ability, or remove the \
+           location from your app config if you do not.`,
```

Comment:
Perhaps we could also add an env var that one can set to ignore this check? On the off chance that someone has a custom processor to consumer these same locations. Something like `BACKSTAGE_CATALOG_SKIP_MISSING_PROCESSORS_CHECK`

Reply:
sure i can add that

Reply:
@Rugvip updated - also snuck in the api report warning-free-ness limitation while i was at it

Reply:
I'm thinking also mention the env var as an escape hatch in this message?

Reply:
Ah of course, I'm rushing too much - added now